File: 452/452.用最少数量的箭引爆气球.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        if len(points) <= 1:
            return len(points)
        
        # 按照区间结束位置进行排序
        points.sort(key=lambda end: end[1])
        
        res = 1
        lastEnd = points[0][1]

        for curStart, curEnd in points :
            if curStart > lastEnd :
                lastEnd = curEnd
                res += 1
            
        return res
        

File: 452/452.用最少数量的箭引爆气球.js
/*
 * @lc app=leetcode.cn id=452 lang=javascript
 *
 * [452] 用最少数量的箭引爆气球
 */

// @lc code=start
/**
 * @param {number[][]} points
 * @return {number}
 */
var findMinArrowShots = function (points) {
  if (points.length <= 1) return points.length;
  points.sort((a, b) => a[1] - b[1]);
  let res = 1;
  let lastEnd = points[0][1];

  points.forEach(([curStart, curEnd]) => {
    if (curStart > lastEnd) {
      lastEnd = curEnd;
      res++;
    }
  });
  return res;
};
// @lc code=end
